tutorials/transformers-text-classification.ipynb

```diff
@@ -0,0 +1,751 @@
+      ],
+      "execution_count": 2,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "WZYyXYB5znkH"
+      },
+      "source": [
+        "## Basic Setup\n",
+        "\n",
+        "Next we will follow the tutorial and load up our dataset and tokenizer to prepocess the data."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "gCVrdHOVgjtH"
+      },
+      "source": [
+        "### Data Preprocessing"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "PQ80tXFPwAnR"
+      },
+      "source": [
+        "from datasets import load_dataset\n",
+        "\n",
+        "raw_datasets = load_dataset(\"imdb\")"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "inP62g7LwV5n"
+      },
+      "source": [
+        "from transformers import AutoTokenizer\n",
+        "\n",
+        "tokenizer = AutoTokenizer.from_pretrained(\"bert-base-cased\")"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "aQgWjFq1yESe"
+      },
+      "source": [
+        "def tokenize_function(examples):\n",
+        "    return tokenizer(examples[\"text\"], padding=\"max_length\", truncation=True)\n",
+        "\n",
+        "tokenized_datasets = raw_datasets.map(tokenize_function, batched=True)"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "xnLbsx2Jhurm"
+      },
+      "source": [
+        "We move towards the end of the tutorial for PyTorch specific instructions. Here we are extracting a larger subset of our original datasets. We also don't need to provide a seed since we seeded everything at the beginning."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "iPQy_FMcxBy3"
+      },
+      "source": [
+        "tokenized_datasets = tokenized_datasets.remove_columns([\"text\"])\n",
+        "tokenized_datasets = tokenized_datasets.rename_column(\"label\", \"labels\")\n",
+        "tokenized_datasets.set_format(\"torch\")\n",
+        "\n",
+        "small_train_dataset = tokenized_datasets[\"train\"].shuffle().select(range(5000))\n",
+        "small_eval_dataset = tokenized_datasets[\"test\"].shuffle().select(range(5000))"
+      ],
+      "execution_count": 6,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "0JvnKjqRiEFP"
+      },
+      "source": [
+        "### Dataloaders"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "APNr5lgsygtw"
+      },
+      "source": [
+        "from torch.utils.data import DataLoader\n",
+        "\n",
+        "train_dataloader = DataLoader(small_train_dataset, shuffle=True, batch_size=8)\n",
```

Comment:
Does `auto_dataloader` work fine here ? Yet, I,didn't try

Reply:
I don't agree that much to use `auto_*` or any distributed things here, as that's not what we want to show in this tutorial, IMO, we should keep things straightforward as possible in this introductory level tutorial

Reply:
Transformers are not cheap models, training is time consuming. Although I understand your meaning, auto helpers are very light and could be a valuable improvement. 

Maybe it could be discussed later.

Reply:
Thank you for your review @sdesrozis but I think I agree with @KickItLikeShika . This tutorial is just meant for introducing Ignite concepts for NLP beginners (that's why I used the most basic example by :hugs: ) and `idist` is still an advanced concept that needs to be discussed in detail like in the CIFAR10 notebook (how to use different configs). I can still introduce the `auto_*` methods without going into detail and just explaining how to run this code as a Jupyter notebook or in Colab. But I feel that this will be incomplete. What do you all think?

Reply:
It makes sense ;)